server/app/services/applicant/question/AddressQuestion.java

```diff
@@ -37,14 +38,24 @@ protected ImmutableSet<QuestionType> validQuestionTypes() {
   }
 
   @Override
-  public ImmutableSet<ValidationErrorMessage> getQuestionErrors() {
-    if (!isAnswered()) {
-      return ImmutableSet.of();
-    }
+  protected ImmutableMap<Path, ImmutableSet<ValidationErrorMessage>> getValidationErrorsInternal() {
+    return ImmutableMap.<Path, ImmutableSet<ValidationErrorMessage>>builder()
+        .put(applicantQuestion.getContextualizedPath(), getQuestionErrors())
```

Comment:
Note: Here we unconditionally put the result of getQuestionErrors, which may be empty. This has the effect of returning a Map that has several keys and no entries. There are presently only two callers of getValidationErrors which account for this. Additionally, in QuestionImpl.java, the implementation of getValidationErrors filters out keys without any validation errors.

Reply:
suggestion (blocking): move this note into a comment in QuestionImpl.java

Reply:
Done!

Reply:
Thanks for the detailed thoughts! Your approach sounds good to me, since ultimately the public function will return correct data either way. And I also think that this approach is less error-prone, since the base class handles all the filtering.

Removing QuestionImpl also sounds good to me!